source/common/common/matchers.cc

```diff
@@ -65,10 +65,16 @@ StringMatcherImpl::StringMatcherImpl(const envoy::type::matcher::v3::StringMatch
     : matcher_(matcher) {
   if (matcher.match_pattern_case() ==
       envoy::type::matcher::v3::StringMatcher::MatchPatternCase::kHiddenEnvoyDeprecatedRegex) {
+    if (matcher.ignore_case()) {
+      PANIC("ignore_case has no effect for regex.");
+    }
     regex_ =
         Regex::Utility::parseStdRegexAsCompiledMatcher(matcher_.hidden_envoy_deprecated_regex());
   } else if (matcher.match_pattern_case() ==
              envoy::type::matcher::v3::StringMatcher::MatchPatternCase::kSafeRegex) {
+    if (matcher.ignore_case()) {
+      PANIC("ignore_case has no effect for safe_regex.");
```

Comment:
This shouldn't be a `PANIC` but throwing an `EnvoyException`, `PANIC` will crash Envoy with bad config, but Envoy should just reject the config.

Reply:
Thanks, updated to use EnvoyException and also added a unit test for this.